### PR TITLE
Bitmex fetch deposit address

### PR DESCRIPTION
**Pull request by samgermain**
I would like to get rid of the nested double quotes in address, but can't remember how we do that

```
2022-12-06T02:19:50.824Z
Node.js: v18.4.0
CCXT v2.2.72
bitmex.fetchDepositAddress (BTC)
2022-12-06T02:19:52.431Z iteration 0 passed in 656 ms

{
  currency: 'BTC',
  address: '"bc1qmex3puyrzn2gduqcnlu70c2uscpyaa9nm2l2j9le2lt2wkgmw33sy7ndjg"',
  tag: undefined,
  network: 'BTC',
  info: '"bc1qmex3puyrzn2gduqcnlu70c2uscpyaa9nm2l2j9le2lt2wkgmw33sy7ndjg"'
}
2022-12-06T02:19:52.431Z iteration 1 passed in 656 ms
```

```
2022-12-06T02:26:58.730Z
Node.js: v18.4.0
CCXT v2.2.72
bitmex.fetchDepositAddress (USDT, [object Object])
2022-12-06T02:27:02.153Z iteration 0 passed in 1600 ms

{
  currency: 'USDT',
  address: '"TNVRj7xS6so5owfLZu6ohpqXvp69mn1bie"',
  tag: undefined,
  network: 'TRX',
  info: '"TNVRj7xS6so5owfLZu6ohpqXvp69mn1bie"'
}
2022-12-06T02:27:02.153Z iteration 1 passed in 1600 ms
```

### CODE REVIEW

File: js/bitmex.js [hunk 1]
```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
...
+         * @param {string} code unified currency code
+         * @param {object} params extra parameters specific to the bitmex api endpoint
+         * @param {network} params.network deposit chain, can view all chains via this.publicGetWalletAssets, default is eth, unless the currency has a default chain within this.options['networks']
+         * @returns {object} an [address structure]{@link https://docs.ccxt.com/en/latest/manual.html#address-structure}
+         */
+        const networkCode = this.safeStringUpper (params, 'network');
+        const networks = this.safeValue (this.options, 'networks', {});
+        let networkId = undefined;
+        if (networkCode === undefined) {
+            networkId = this.safeStringLower (networks, code, 'eth');
```
Comment (carlosmiei) [inline js/bitmex.js:2714]: this is very error-prone, why not require the network in `params` explicitly? 

For instance, `bitmex.fetchDepositAddress (USDT)` fails with `bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}`

Comment (samgermain) [reply js/bitmex.js:2714]: it just feels kinda unnecessary to require it, 80% of the time it's ETH, then 19% of the time, it's the same as the currency code

Comment (carlosmiei) [reply js/bitmex.js:2714]: @samgermain I would not be so sure about it, most people use the network with the lower fees, and as I said, something so simple as `bitmex.fetchDepositAddress (USDT)` fails, so I think we should change it

File: js/bitmex.js [hunk 2]
```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
...
+        const currency = this.currency (code);
+        let currencyId = currency['id'];
+        const idLength = currencyId.length;
+        currencyId = currencyId.slice (0, idLength - 1) + currencyId.slice (idLength - 1, idLength).toLowerCase ();  // make the last letter lowercase
+        params = this.omit (params, 'network');
+        const request = {
+            'currency': currencyId,
+            'network': networkId,
+        };
+        const response = await this.privateGetUserDepositAddress (this.extend (request, params));
```
Comment (carlosmiei) [inline js/bitmex.js:2727]: @samgermain something like `fetchDepositAddress(USDT, {"network":"ERC20"})` fails. Can you take a look pls?

```
bitmex.fetchDepositAddress (USDT, [object Object])
BadRequest bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}
---------------------------------------------------
[BadRequest] bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}

    at handleErrors               js/bitmex.js:2767                     throw new BadRequest (feedback);                                                
    at                            js/base/Exchange.js:599               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections  node:internal/process/task_queues:95                                                                                  
    at async timeout              js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                     js/base/Exchange.js:1967              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                    js/base/Exchange.js:1971              return await this.fetch2 (path, api, method, params, headers, body, config, con…
    at fetchDepositAddress        js/bitmex.js:2727                     const response = await this.privateGetUserDepositAddress (this.extend (request,…
    at async run                  examples/js/cli.js:281                const result = await exchange[methodName] (... args)                            

bitmex {"error":{"message":"Invalid network for currency","name":"HTTPError"}}
```

Comment (carlosmiei) [reply js/bitmex.js:2727]: @samgermain can you check this comment pls

Comment (samgermain) [reply js/bitmex.js:2727]: Should be good now

Comment (carlosmiei) [reply js/bitmex.js:2727]: @samgermain will check it thanks

File: js/bitmex.js [hunk 3]
```diff
@@ -2666,6 +2697,46 @@ module.exports = class bitmex extends Exchange {
...
+        const networkCode = this.safeStringUpper (params, 'network');
+        if (networkCode === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDepositAddress requires params["network"]');
+        }
+        const networks = this.safeValue (this.options, 'networks', {});
+        const networkId = this.safeStringLower (networks, networkCode, networkCode);
+        const currency = this.currency (code);
+        let currencyId = currency['id'];
+        const idLength = currencyId.length;
+        currencyId = currencyId.slice (0, idLength - 1) + currencyId.slice (idLength - 1, idLength).toLowerCase ();  // make the last letter lowercase
```
Comment (carlosmiei) [inline js/bitmex.js:2721]: @ttodua tagging you so we don't forget after your merge, we can remove this logic because the id will be correct. But don't worry we can do this in a separate PR

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
p bitmex fetchDepositAddress "USDT" '{"network":"ERC20"}'
Python v3.9.7
CCXT v2.4.18
bitmex.fetchDepositAddress(USDT,{'network': 'ERC20'})
{'address': '0x049Fb5d39F51C43e6c2088Dda83B7E200d26A5e6',
 'currency': 'USDT',
 'info': '"0x049Fb5d39F51C43e6c2088Dda83B7E200d26A5e6"',
 'network': 'ERC20',
 'tag': None}
```
```
p bitmex fetchDepositAddress "USDT" '{"network":"TRC20"}'
Python v3.9.7
CCXT v2.4.18
bitmex.fetchDepositAddress(USDT,{'network': 'TRC20'})
{'address': 'TKwfNxUAFqvEjX9SknDLFJxUgCtKJoxYaD',
 'currency': 'USDT',
 'info': '"TKwfNxUAFqvEjX9SknDLFJxUgCtKJoxYaD"',
 'network': 'TRX',
 'tag': None}
```

**Review by carlosmiei — APPROVED** [review]

File: js/bitmex.js (hunk 3 above)
Comment (samgermain) [reply js/bitmex.js:2720]: The ids were usually something like `ETH`, but the api endpoint usually requires something like `ETh` for some reason, and it was only this endpoint

Comment (ttodua) [reply js/bitmex.js:2720]: @carlosmiei I think my bitmex PR includes all fixes for bitmex, related to currencies & networks & markets.

Comment (samgermain) [reply js/bitmex.js:2720]: > @carlosmiei I think my bitmex PR includes all fixes for bitmex, related to currencies & networks & markets.

This endpoint specifically requires all uppercase and the last letter to be lowercase, but other ones don't, so I don't see how you could have written a fix for this issue
